ocean_provider/routes/compute.py

```diff
@@ -342,3 +344,73 @@ def computeStart():
         )
     except (ValueError, KeyError, Exception) as e:
         return service_unavailable(e, data, logger)
+
+@services.route("/computeResult", methods=["GET"])
+@validate(ComputeGetResult)
+def ComputeResult():
+    """Allows download of asset data file.
+
+    ---
+    tags:
+      - services
+    consumes:
+      - application/json
+    parameters:
+      - name: consumerAddress
+        in: query
+        description: The consumer address.
+        required: true
+        type: string
+      - name: jobId
+        in: query
+        description: JobId
+        required: true
+        type: string
+      - name: index
+        in: query
+        description: Result index
+        required: true
+      - name: signature
+        in: query
+        description: Signature of (jobId+index+consumerAddress) to verify that the consumer has rights to download the result
+    responses:
+      200:
+        description: Content of the result
+      400:
+        description: One of the required attributes is missing.
+      404:
+        description: Result not found
+      503:
+        description: Service Unavailable
+    """
+    data = get_request_data(request)
+    logger.info(f"computeResult endpoint called. {data}")
+    url = get_compute_endpoint()
+    msg_to_sign = f"{data.get('jobId')}{data.get('index')}{data.get('consumerAddress')}"
+    # we sign the same message as consumer does, but using our key
+    providerSignature = sign_message(msg_to_sign, provider_wallet)
+    result_url = (
```

Comment:
please use PreparedRequest to form the URL, as seen here:

https://github.com/oceanprotocol/provider/blob/ab2c9f8e3bc9f556d19a03cc8795231cd72be384/ocean_provider/utils/url.py#L214

Forming strings like this is hard to maintain and difficult to read.